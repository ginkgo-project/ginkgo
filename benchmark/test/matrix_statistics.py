#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="matrix_statistics.simple.stdout",
    expected_stderr="matrix_statistics.simple.stderr",
)
