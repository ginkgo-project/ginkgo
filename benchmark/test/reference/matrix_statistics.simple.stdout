
[
    {
        "size": 125,
        "stencil": "7pt",
        "problem": {
            "rows": 125,
            "columns": 125,
            "nonzeros": 725,
            "row_distribution": {
                "min": 4,
                "q1": 5.0,
                "median": 6.0,
                "q3": 6.0,
                "max": 7,
                "mean": 5.8,
                "variance": 0.7199999999999992,
                "skewness": -0.23570226039551892,
                "kurtosis": 2.388888888888889,
                "hyperskewness": -1.741577812922432,
                "hyperflatness": 7.762345679012379
            },
            "col_distribution": {
                "min": 4,
                "q1": 5.0,
                "median": 6.0,
                "q3": 6.0,
                "max": 7,
                "mean": 5.8,
                "variance": 0.7199999999999992,
                "skewness": -0.23570226039551892,
                "kurtosis": 2.388888888888889,
                "hyperskewness": -1.741577812922432,
                "hyperflatness": 7.762345679012379
            }
        }
    }
]
